server/event/dispatch/operation.go

```diff
@@ -129,6 +137,45 @@ func (o *Operation) dispatch(wfeb wfv1.WorkflowEventBinding, nameSuffix string)
 	return nil, nil
 }
 
+func (o *Operation) populateWorkflowMetadata(wf *wfv1.Workflow, metadata *metav1.ObjectMeta) error {
+	if len(metadata.Name) > 0 {
+		evalName, err := o.evaluateStringExpression(metadata.Name, "name")
+		if err != nil {
+			return err
+		}
+		wf.SetName(evalName)
+	}
+	for labelKey, labelValue := range metadata.Labels {
+		evalLabel, err := o.evaluateStringExpression(labelValue, fmt.Sprintf("label \"%s\"", labelKey))
+		if err != nil {
+			return err
+		}
+		wf.Labels[labelKey] = evalLabel
+	}
+	for annotationKey, annotationValue := range metadata.Annotations {
+		evalAnnotation, err := o.evaluateStringExpression(annotationValue, fmt.Sprintf("annotation \"%s\"", annotationKey))
+		if err != nil {
+			return err
+		}
+		wf.Annotations[annotationKey] = evalAnnotation
+	}
+	return nil
+}
+
+func (o *Operation) evaluateStringExpression(statement string, errorInfo string) (string, error) {
+	result, err := expr.Eval(statement, o.env)
+	if err != nil {
+		return "", fmt.Errorf("failed to evaluate workflow %s expression: %w", errorInfo, err)
+	}
+
+	switch v := result.(type) {
```

Comment:
minor - `v, ok := result.(string)` is a more conventional way of doing this

Reply:
Fixed. Thanks for showing me this.